content/blog/2019/secure-egress-control-istio/index.md

```diff
@@ -0,0 +1,323 @@
+
+Let's examine solutions for egress traffic control that existed before Istio in the following section.
+
+## Existing solutions for egress traffic control
+
+The most natural solution for egress traffic control is
+[Kubernetes Network Policies](https://kubernetes.io/docs/concepts/services-networking/network-policies/). Using
+Kubernetes Network Policies, cluster operators can specify which external services can be accessed by which pods. The
+pods may be identified by pod labels, namespace labels, or by IP ranges. The external services can be specified by IP
+ranges: Kubernetes Network Policies are not DNS-aware. The requirement 1 is satisfied since any TCP traffic can be
+controlled by Kubernetes Network Policies. The requirements 3 and 4 are satisfied partially: the policies can be
+specified per cluster or per pod, however the external services cannot be identified by domain names.
+The requirement 5 is satisfied if the attackers are not able to break from a malicious container into the Kubernetes
+node and to interfere with the implementation of the Kubernetes Network Policies inside the node.
+The requirement 6 is satisfied as well: there is no need to change the code or the
+container environment. We can say that Kubernetes Network Policies provide transparent, Kubernetes-aware egress traffic
+control, which is not DNS-aware.
+
+Another approach that predates Kubernetes network policies is a **DNS-aware egress proxy/firewall**. In this
+approach, applications are configured to direct the traffic to the proxy and to use some proxy protocol, e.g.
+[SOCKS](https://en.wikipedia.org/wiki/SOCKS).
+Since the applications must be configured, this solution is not transparent. Moreover, egress proxies are not
+Kubernetes-aware, since neither pod labels nor pod service account are known to the egress proxy. Such egress proxies
+cannot fulfill the requirement 4, that is they cannot enforce policies by source if the source is specified by a
+Kubernetes artifact. The egress proxies can fulfill the requirements 1, 2, 3 and 5, but not the requirement 4 and 6.
+They are DNS-aware, but not transparent and not Kubernetes-aware.
+
+Let me explain Istio egress traffic control next.
+
+## Egress traffic control by Istio
+
+To implement secure egress traffic control in Istio, you must
+[direct TLS traffic to external services through an egress gateway](/docs/examples/advanced-gateways/egress-gateway/#egress-gateway-for-https-traffic).
+(To support wildcard domains, you must create
+[a custom version of an egress gateway](/docs/examples/advanced-gateways/wildcard-egress-hosts/)). Alternatively, you
+can [direct HTTP traffic through an egress gateway](/docs/examples/advanced-gateways/egress-gateway/#egress-gateway-for-http-traffic)
+and [let the egress gateway perform TLS origination](/docs/examples/advanced-gateways/egress-gateway-tls-origination/#perform-tls-origination-with-an-egress-gateway).
+
+In all the cases you have to apply some
```

Comment:
the cases -> cases

Reply:
Fixed by https://github.com/istio/istio.io/pull/3179/commits/326adf8be1f5021bff78c5a16286a8c8f61d7ef1